compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3608,11 +3609,34 @@ private boolean isInvalidNeverField(BField lhsField, BRecordType rhsType) {
 
     private BAttachedFunction getMatchingInvokableType(List<BAttachedFunction> rhsFuncList, BAttachedFunction lhsFunc,
                                                        Set<TypePair> unresolvedTypes) {
-        return rhsFuncList.stream()
+        BAttachedFunction matchingFunction = rhsFuncList.stream()
                 .filter(rhsFunc -> lhsFunc.funcName.equals(rhsFunc.funcName))
                 .filter(rhsFunc -> isFunctionTypeAssignable(rhsFunc.type, lhsFunc.type, unresolvedTypes))
                 .findFirst()
                 .orElse(null);
+        
+        // Todo: We could include this logic in `isFunctionTypeAssignable` if we have `resourcePathType` information in
+        // `BInvokableType`
+        if (matchingFunction == null || 
+                !Symbols.isResource(matchingFunction.symbol) || !Symbols.isResource(lhsFunc.symbol)) {
+            return matchingFunction;
+        }
+
+        List<BType> lhsFuncResourcePathTypes = ((BResourceFunction) lhsFunc).resourcePathType.tupleTypes;
```

Comment:
Don't we need these changes at runtime too?

Reply:
added in [64953a6](https://github.com/ballerina-platform/ballerina-lang/pull/37494/commits/64953a6b9690cf3ea747fc453cb01bc1b176428c)